ros/src/common/cmake/autoware_build_flags/cmake/autoware_build_flags-extras.cmake

```diff
@@ -11,22 +11,25 @@ else()
   set(CMAKE_CXX_STANDARD 11)
 endif()
 
-option(ENABLE_CUDA "Enable_CUDA" FALSE)
+message(STATUS "CUDA compilation status: $ENV{AUTOWARE_COMPILE_WITH_CUDA}.")
 
-if (ENABLE_CUDA)
-  find_package(CUDA REQUIRED)
-  find_package(Eigen3 REQUIRED)
+macro(AW_CHECK_CUDA)
+  if ($ENV{AUTOWARE_COMPILE_WITH_CUDA})
+    find_package(CUDA REQUIRED)
+    find_package(Eigen3 REQUIRED)
 
-  if(${CUDA_VERSION} VERSION_GREATER "9.1"
-        AND ${CMAKE_VERSION} VERSION_LESS "3.12.3")
-    unset(CUDA_cublas_device_LIBRARY CACHE)
-    set(CUDA_cublas_device_LIBRARY ${CUDA_cublas_LIBRARY})
-    set(CUDA_CUBLAS_LIBRARIES ${CUDA_cublas_LIBRARY})
+    if(${CUDA_VERSION} VERSION_GREATER "9.1"
+          AND ${CMAKE_VERSION} VERSION_LESS "3.12.3")
+      unset(CUDA_cublas_device_LIBRARY CACHE)
+      set(CUDA_cublas_device_LIBRARY ${CUDA_cublas_LIBRARY})
+      set(CUDA_CUBLAS_LIBRARIES ${CUDA_cublas_LIBRARY})
+    endif()
+    if ("$ENV{ROS_DISTRO}" STREQUAL "melodic" AND ${EIGEN3_VERSION_STRING} VERSION_LESS "3.3.7")
+      message(FATAL_ERROR "GPU support on Melodic requires Eigen version>= 3.3.7")
+    endif()
+    set(USE_CUDA ON)
+  else()
+    message(WARNING "CUDA support is disabled. Set the env var AUTOWARE_COMPILE_WITH_CUDA=1 and recompile to enable it")
```

Comment:
"Set the env var AUTOWARE_COMPILE_WITH_CUDA=1" -> "Set the AUTOWARE_COMPILE_WITH_CUDA environment variable"

Reply:
done!